diff -Naur rsaref2/install/unix/makefile rsaref2.work/install/unix/makefile
--- rsaref2/install/unix/makefile	Mon May  9 15:18:57 1994
+++ rsaref2.work/install/unix/makefile	Sat Nov 17 21:32:15 2001
@@ -30,8 +30,11 @@
 # The location of the demo source directory.
 RDEMODIR = ../../rdemo/
 
-all : rdemo dhdemo
+all : clean rdemo dhdemo
 
+clean : 
+	rm -f *.o rdemo dhdemo librsaref2.so rsaref.a
+	 
 rdemo : rdemo.$(O) $(RSAREFLIB)
 	cc -o $@ rdemo.$(O) $(RSAREFLIB)
 
diff -Naur rsaref2/install/unix/makefile.shared rsaref2.work/install/unix/makefile.shared
--- rsaref2/install/unix/makefile.shared	Thu Jan  1 00:00:00 1970
+++ rsaref2.work/install/unix/makefile.shared	Sat Nov 17 21:33:08 2001
@@ -0,0 +1,57 @@
+# This is a makefile for UNIX
+
+# extension for object files
+O = o
+
+# commands
+CC = cc
+LIB = ar
+
+# name of temporary library script
+TEMPFILE = $(TEMP)/temp.mak
+
+# standard include directory
+STDINCDIR=/usr/include
+
+# The places to look for include files (in order).
+INCL =  -I. -I$(RSAREFDIR) -I$(STDINCDIR)
+
+# name of main executable to build
+PROG = all
+
+# Normal C flags.
+CFLAGS = $(INCL) -O -fPIC -g -c -DPROTOTYPES=1
+MFLAGS = -I. -I$(RSAREFDIR)
+
+# The location of the common source directory.
+RSAREFDIR = ../../source/
+RSAREFLIB = librsaref2.so
+
+# The location of the demo source directory.
+RDEMODIR = ../../rdemo/
+
+all : clean rdemo dhdemo
+
+clean : 
+	rm -f *.o rdemo dhdemo librsaref2.so rsaref.a
+	
+rdemo : rdemo.$(O) $(RSAREFLIB)
+	cc -o $@ rdemo.$(O) $(RSAREFLIB)
+
+dhdemo : dhdemo.$(O) $(RSAREFLIB)
+	cc -o $@ dhdemo.$(O) $(RSAREFLIB)
+
+OBJS = desc.$(O) digit.$(O) md2c.$(O) md5c.$(O) nn.$(O) prime.$(O)\
+  rsa.$(O) r_encode.$(O) r_dh.$(O) r_enhanc.$(O) r_keygen.$(O) r_random.$(O)\
+  r_stdlib.$(O)
+
+$(RSAREFLIB) : $(OBJS)
+	$(CC) -g -shared -Wl,-soname,$(RSAREFLIB) -o $(RSAREFLIB) $(OBJS) -lc
+
+rdemo.$(O) : $(RDEMODIR)rdemo.c $(RSAREFDIR)global.h $(RSAREFDIR)rsaref.h
+	$(CC) $(CFLAGS) $(RDEMODIR)rdemo.c
+
+dhdemo.$(O) : $(RDEMODIR)dhdemo.c $(RSAREFDIR)global.h $(RSAREFDIR)rsaref.h
+	$(CC) $(CFLAGS) $(RDEMODIR)dhdemo.c
+
+include $(RSAREFDIR)targets.mak
